sdn_tests/testplans/Stress_Latency_Test_Plan.md

```diff
@@ -0,0 +1,385 @@
+# Overview
+
+This document aims to outline the approach for testing the stress and latency of gNMI GET/SET operations supported by the gNMI protocol as part of GPINs OpenConfig end-to-end testing. 
+
+# Background
+
+With SONiC as the network operating system (NOS) for GPINS, gNMI is responsible for monitoring, streaming telemetry, and configuration management.  Broadcom's Unified Management Framework (UMF) provides gNMI streaming telemetry based on the standard OpenConfig model.
+
+The gNMI client can invoke a GET,  and/or SET request. The SET operations can be further categorized into UPDATE, REPLACE, and DELETE operations.
+
+# gNMI Protocol Testing
+
+The current approach to test the gNMI protocol features is to leverage the existing or planned end-to-end tests to be written for the gNMI paths. The idea is to reduce the duplication of test cases and use the ones already planned for verifying the paths to test the protocol features at the same time.\
+For example, any of the tests that use a GET/SET request for any of the config leafs can be extended to also test the config subtrees. \
+As to the question of which of the path tests are to be used for which feature test, that is still TBD based on the completion and review of all the path E2E tests.
+
+# Purpose
+
+The purpose of the gNMI stress and latency tests is to test the robustness of the gNMI server and capture various system metrics as a report at the end of the testing. The validation for the correctness of the data is out of the scope since the payload in the response ( data ) is application specific. The application (feature owner) can consume the report and set the threshold for the gNMI operations based on the platforms. 
+
+# E2E Stress Test Cases
+
+The purpose of the gNMI stress test is to validate the robustness of the telemetry service and overall performance. The stress testing will not validate the correctness of the data that is being received from the server.  Since the gNMI performance is dependent on the platform, the tests will not enforce the threshold to be validated as part of the validation, instead the tests will gather the metrics and produce the performance report at the end of the tests. The final report will be analyzed and checked against the platform specific requirements. There will also be periodic sanity checks during the test to capture test failures early.
+
+The robustness is validated through the following methods
+
+-   Monitoring the CPU load for the gNMI server
```

Comment:
This line can be removed. I don't think the tests currently monitor the CPU load.